From: Henri Verbeet <hverbeet@codeweavers.com>
Subject: [PATCH vkd3d 4/6] vkd3d-shader: Use full write mask for variables for array built-ins.
Message-Id: <20191024172817.5662-4-hverbeet@codeweavers.com>
Date: Thu, 24 Oct 2019 20:58:15 +0330

From: Philip Rebohle <philip.rebohle@tu-dortmund.de>

Private variables are always vec4, so using a sparse write mask here
will lead to invalid code being generated when accessing the variable.

Signed-off-by: Philip Rebohle <philip.rebohle@tu-dortmund.de>
Signed-off-by: Henri Verbeet <hverbeet@codeweavers.com>
---
This supersedes patch 171405.

 libs/vkd3d-shader/spirv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libs/vkd3d-shader/spirv.c b/libs/vkd3d-shader/spirv.c
index 6a3eb10..936b014 100644
--- a/libs/vkd3d-shader/spirv.c
+++ b/libs/vkd3d-shader/spirv.c
@@ -4320,7 +4320,7 @@ static void vkd3d_dxbc_compiler_emit_output(struct vkd3d_dxbc_compiler *compiler
     storage_class = SpvStorageClassOutput;
 
     if ((use_private_variable = builtin && builtin->spirv_array_size))
-        write_mask = dst->write_mask;
+        write_mask = VKD3DSP_WRITEMASK_ALL;
     else if (get_shader_output_swizzle(compiler, signature_element->register_index) != VKD3D_NO_SWIZZLE
             || needs_private_io_variable(shader_signature, signature_element->register_index,
                     builtin, &output_component_count, &write_mask)

-- 
2.11.0

